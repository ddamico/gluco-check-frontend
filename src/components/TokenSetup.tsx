import React from "react";
import {
  Button,
  makeStyles,
  Paper,
  Step,
  StepContent,
  StepLabel,
  Stepper,
  Typography,
} from "@material-ui/core";
import { useTranslation } from "react-i18next";

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 400,
  },
  button: {
    marginTop: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  actionsContainer: {
    marginBottom: theme.spacing(2),
  },
  resetContainer: {
    padding: theme.spacing(3),
  },
}));

interface TokenStep {
  label: string;
  description: string;
}

export default function TokenSetup() {
  const { t } = useTranslation();
  const classes = useStyles();
  const [activeStep, setActiveStep] = React.useState(0);

  const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);
  };

  const handleBack = () => {
    console.log("handleback");
    setActiveStep((prevActiveStep) => prevActiveStep - 1);
  };

  const handleReset = () => {
    setActiveStep(0);
  };

  const tokenSteps: TokenStep[] =
    t("tokenDialog.steps", {
      returnObjects: true,
    }) ?? [];

  const steps = tokenSteps.map((step) => step.label);
  const stepContent = tokenSteps.map((step) => step.description);

  return (
    <div className={classes.root}>
      <Stepper activeStep={activeStep} orientation="vertical">
        {steps.map((label, index) => (
          <Step key={label}>
            <StepLabel>{label}</StepLabel>
            <StepContent>
              <Typography>{stepContent[index]}</Typography>
              <div className={classes.actionsContainer}>
                <div>
                  <Button
                    disabled={activeStep === 0}
                    onClick={handleBack}
                    className={classes.button}
                    data-testid="token-stepper-back"
                  >
                    Back
                  </Button>
                  <Button
                    variant="contained"
                    color="primary"
                    onClick={handleNext}
                    className={classes.button}
                    data-testid="token-stepper-next"
                  >
                    {activeStep === steps.length - 1 ? "Finish" : "Next"}
                  </Button>
                </div>
              </div>
            </StepContent>
          </Step>
        ))}
      </Stepper>
      {activeStep === steps.length && (
        <Paper square elevation={0} className={classes.resetContainer}>
          <Typography>All steps completed - you&apos;re finished</Typography>
          <Button
            onClick={handleReset}
            className={classes.button}
            data-testid="token-stepper-reset"
          >
            Reset
          </Button>
        </Paper>
      )}
    </div>
  );
}
