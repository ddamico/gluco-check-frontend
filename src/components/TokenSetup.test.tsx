import React from "react";
import {
  cleanup,
  render,
  waitFor,
  screen,
  prettyDOM,
} from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import TokenSetup from "./TokenSetup";
import userEvent from "@testing-library/user-event";

expect.extend(toHaveNoViolations);

afterEach(() => {
  jest.resetAllMocks();
  cleanup();
});

const mockTokenSteps = [
  {
    label: "Step 1 label",
    description: "Step 1 description",
  },
  {
    label: "Step 2 label",
    description: "Step 2 description",
  },
  {
    label: "Step 3 label",
    description: "Step 3 description",
  },
];

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i, options) => {
        if (i === "tokenDialog.steps") {
          return mockTokenSteps;
        }
        return i;
      }),
    };
  },
}));

describe("Container component", () => {
  it("renders the component", () => {
    const { container } = render(<TokenSetup />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("can step through the component's steps", async () => {
    const { container } = render(<TokenSetup />);
    let nextButton: HTMLElement | null, backButton: HTMLElement | null;

    // can traverse stepper forward
    for (let n = 0; n <= mockTokenSteps.length; n++) {
      nextButton = await screen.queryByTestId("token-stepper-next");
      if (nextButton) {
        await waitFor(() => {
          userEvent.click(nextButton);
        });
        expect(container.firstChild).toMatchSnapshot(`Step ${n}, forward`);
      }
    }

    // can traverse stepper backward
    for (let i = 0; i <= mockTokenSteps.length; i++) {
      backButton = await screen.queryByTestId("token-stepper-back");
      if (backButton) {
        await waitFor(() => {
          userEvent.click(backButton);
        });
        expect(container.firstChild).toMatchSnapshot(`Step ${i}, backward`);
      }
    }
  });

  it("has no axe violations", async () => {
    const { container } = render(<TokenSetup />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
