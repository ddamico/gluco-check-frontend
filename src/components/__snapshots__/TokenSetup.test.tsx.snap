// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component can step through the component's steps: Step 0, forward 1`] = `
<div
  class="makeStyles-root-5"
>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 1 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-active"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 2 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="2"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style="min-height: 0px; height: auto; transition-duration: 0ms;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Step 2 description
              </p>
              <div
                class="makeStyles-actionsContainer-7"
              >
                <div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-6"
                    data-testid="token-stepper-back"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Back
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-6 MuiButton-containedPrimary"
                    data-testid="token-stepper-next"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Next
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 3 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        disabled=""
        icon="3"
      />
    </div>
  </div>
</div>
`;

exports[`Container component can step through the component's steps: Step 1, forward 1`] = `
<div
  class="makeStyles-root-5"
>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 1 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 2 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="2"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-active"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 3 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        icon="3"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style="min-height: 0px; height: auto; transition-duration: 0ms;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Step 3 description
              </p>
              <div
                class="makeStyles-actionsContainer-7"
              >
                <div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-6"
                    data-testid="token-stepper-back"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Back
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-6 MuiButton-containedPrimary"
                    data-testid="token-stepper-next"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Finish
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Container component can step through the component's steps: Step 2, forward 1`] = `
<div
  class="makeStyles-root-5"
>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 1 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 2 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="2"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 3 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        icon="3"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition"
          style="min-height: 0px; height: 0px; transition-duration: 0ms;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Step 3 description
              </p>
              <div
                class="makeStyles-actionsContainer-7"
              >
                <div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-6"
                    data-testid="token-stepper-back"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Back
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-6 MuiButton-containedPrimary"
                    data-testid="token-stepper-next"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Next
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root makeStyles-resetContainer-8 MuiPaper-elevation0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      All steps completed - you're finished
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-6"
      data-testid="token-stepper-reset"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Reset
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Container component can step through the component's steps: Step 3, forward 1`] = `
<div
  class="makeStyles-root-5"
>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 1 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 2 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="2"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 3 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        icon="3"
      />
    </div>
  </div>
</div>
`;

exports[`Container component renders the component 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 1 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Step 1 description
              </p>
              <div
                class="makeStyles-actionsContainer-3"
              >
                <div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2 Mui-disabled Mui-disabled"
                    data-testid="token-stepper-back"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Back
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
                    data-testid="token-stepper-next"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Next
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 2 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        disabled=""
        icon="2"
      />
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Step 3 label
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        disabled=""
        icon="3"
      />
    </div>
  </div>
</div>
`;
